src/components/Table/Table.tsx

```diff
@@ -1,27 +1,85 @@
-import { Table as MantineTable, TextInput } from "@mantine/core";
-import { type ChangeEvent, type ReactNode, useMemo, useState } from "react";
-import { IconSortAscending, IconSortDescending } from "@tabler/icons-react";
+import {
+  Table as MantineTable,
+  TextInput,
+  Pagination as MantinePagination,
+  createStyles,
+  Box,
+} from "@mantine/core";
+import {
+  type ChangeEvent,
+  type ReactNode,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import {
+  IconArrowBarToLeft,
+  IconArrowBarToRight,
+  IconArrowLeft,
+  IconArrowRight,
+  IconGripHorizontal,
+  IconSortAscending,
+  IconSortDescending,
+} from "@tabler/icons-react";
 import { omit } from "lodash/fp";
-import { filterData, getNextSortValue, sortData } from "../../utils/utils.ts";
+import {
+  filterData,
+  getMaximumPages,
+  getNextSortValue,
+  sortData,
+} from "../../utils/utils.ts";
+import { useElementSize } from "@mantine/hooks";
 
 export interface Column<TData> {
   columnName: string;
   exactMatch: boolean;
   header: string;
   cellRenderer?: (row: TData) => ReactNode;
+  isFilterable: boolean;
 }
 
+interface Pagination {
+  pageSize: number;
+}
 interface TableProps<TData> {
   data: TData[];
   columns: Array<Column<TData>>;
+  pagination: Pagination;
 }
 
 export type Sort = Record<string, "asc" | "desc" | null>;
 export type Filter = Record<string, string>;
 
 export function Table<T extends { id: number }>(props: TableProps<T>) {
+  const { classes } = useStyles();
+
+  const { ref: rowRef, height: rowHeight } = useElementSize();
+
+  const [windowHeight, setWindowDimensions] = useState(window.innerHeight);
```

Comment:
In your specific case, the height of the window is also the height of the whole table. But in real life scenario, the table would probably sit inside an app, with its own header, sidebar and etc. It's better to calculate stuff based on the Table's height, instead of the window